File: php/testcode/headerinj_001.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_headerinj_location
function headerinj001(BenchmarkRequest $req): BenchmarkResponse {
    $input = $req->param('url');
    header("Location: " . $input); // vuln-code-snippet vuln-line php_headerinj_location
    return BenchmarkResponse::ok('Redirecting...');
}
// vuln-code-snippet end php_headerinj_location

File: ruby/testcode/headerinj_001.rb
require_relative 'shared'

# vuln-code-snippet start ruby_headerinj_location
def redirect_to_url(req)
  input = req.param('url')
  BenchmarkResponse.new(302, '', { 'Location' => input })  # vuln-code-snippet vuln-line ruby_headerinj_location
end
# vuln-code-snippet end ruby_headerinj_location
